vllm/config/__init__.py

```diff
@@ -2368,6 +2368,75 @@ def __repr__(self) -> str:
         return f"SpeculativeConfig({method=}, {model=}, {num_spec_tokens=})"
 
 
+@config
+@dataclass
+class AFDConfig:
+    """Configuration for AFD (Attention FFN Disaggregation) distributed
+    computation."""
+
+    afd_connector: str = "dummy"
+    """The AFD connector for vLLM to communicate between attention and FFN
+    nodes. Available connectors: 'dummy', 'stepmesh'"""
+
+    afd_role: Literal["attention", "ffn"] = "attention"
+    """Role of this vLLM instance in AFD. 'attention' for attention workers,
+    'ffn' for FFN servers."""
+
+    afd_port: int = 1239
+    """Port number for stepmesh parameter server communication."""
+
+    afd_host: str = "127.0.0.1"
+    """Host address for stepmesh parameter server communication."""
+
+    num_afd_stages: int = 3
+    """Number of pipeline stages for stage parallelism."""
+
+    num_attention_servers: int = 1
+    """Number of attention servers."""
+
+    num_ffn_servers: int = 1
+    """Number of FFN servers."""
+
+    afd_server_rank: int = 0
+    """Rank of this AFD server."""
+
+    afd_extra_config: dict[str, Any] = field(default_factory=dict)
+    """Extra configuration for specific AFD connectors."""
+
```

Comment:
When a user develops a custom AFD Connector, they might want to use it without modifying the VLLM code. Therefore, it's possible to add a feature to load the Connector module from an external source. This feature could be submitted after the PR is merged, similar to `kv_connector_module_path`.

I create a issue record https://github.com/vllm-project/vllm/issues/27872